MELOSYS-4001 - Vis norske arbeidsgivere fra SED

sigbjorn-myhre opened the pull request:
Norske arbeidsgivere fra SED blir mappet til `behandlingsgrunnlag.norskeArbeidsgivere`. Legger disse inn blant resten av "andre arbeidsforhold i Norge"

sigbjorn-myhre committed:
Ikke kast exception for 404 ved henting av kontaktopplysninger

sigbjorn-myhre committed:
Mulighet for validering ved mount i orgnrinput.js

sigbjorn-myhre committed:
validerVedMount -> hentVedMount i orgnrinput.js samt feilmeldinger som props

sigbjorn-myhre committed:
Vis norske arbeidsgivere fra SED i andreArbeidsforholdNorge

JohanSakshaug commented on src/felleskomponenter/kontaktopplysninger.js:
```diff
@@ -210,18 +229,20 @@ const mapStateToProps = state => ({
+    hentOrg={fagsakerApi.hentOrg}
+    hentKontaktopplysninger={fagsakerApi.hentKontaktopplysninger}
+    lagreKontaktopplysninger={fagsakerApi.lagreKontaktopplysninger}
+    slettKontaktopplysninger={fagsakerApi.slettKontaktOpplysninger}
```
Jeg tror egentlig du kan fjerne disse propsene helt og bare bruke `Api.Fagsaker.kontaktopplysninger` direkte i komponenten, hvis du vil. Dette var ikke en veldig elegant løsning av meg...

JohanSakshaug commented on src/felleskomponenter/paneler/andreArbeidsforholdNorge/arbeidsforholdNorgeListe/orgnrinput.js:
```diff
@@ -9,17 +9,27 @@ const Orgnrinput = ({
+  }, [validerVedMount]);
+
+  const leggTilOrg = async tillagtOrgnr => {
+    valider(tillagtOrgnr);
```
`return` hvis valideringen feiler?

sigbjorn-myhre replied:
Jepp! 👍  Glemte den da jeg refaktorerte

JohanSakshaug commented on src/felleskomponenter/paneler/andreArbeidsforholdNorge/andreArbeidsforholdNorge.test.js:
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import * as Nav from '../../../utils/navFrontend';
 
-import { AndreArbeidsforholdNorge } from './andreArbeidsforholdNorge';
+import { AlleNorskeArbeidsgivere, AndreArbeidsforholdNorge } from './andreArbeidsforholdNorge';
```
Slenger du på en test av `NorskeArbeidsgivereSed` også? :)

JohanSakshaug commented on src/felleskomponenter/paneler/andreArbeidsforholdNorge/andreArbeidsforholdNorge.test.js:
```diff
@@ -21,12 +21,24 @@ describe('AndreArbeidsforholdNorge', () => {
+    };
+  });
 
   it('viser to ArbeidsforholdNorgeListe', () => {
```
to -> en

sigbjorn-myhre committed:
Fjern props og wrapper i kontaktopplysninger.js som ikke trengs

sigbjorn-myhre committed:
QA MELOSYS-4001

- Legger til tester for NorskeArbeidsgivereSed
- Return dersom validering feiler i orgnrinput.js

JohanSakshaug commented on src/felleskomponenter/kontaktopplysninger.js:
```diff
@@ -229,21 +222,4 @@ const mapStateToProps = state => ({
-  />
-);
-
-export default connect(mapStateToProps)(KontaktOpplysningerWrapper);
```
🎉

sigbjorn-myhre commented:
Venter med merge til test er OK

sigbjorn-myhre committed:
Merge branch 'master' into MELOSYS-4001_vis_norske_arbeidsgivere_fra_sed

sigbjorn-myhre committed:
Fikser en test etter merge

sigbjorn-myhre committed:
Fjern bruk av norskeArbeidsgivere

sigbjorn-myhre committed:
Bruker ekstraArbeidsgivere for å vise norske arbeidsgivere fra SED

sigbjorn-myhre committed:
Fjern tilleggsdata fra behandlingsgrunnlag

JohanSakshaug commented on src/ducks/behandlingsgrunnlag/operations.js:
```diff
@@ -11,23 +11,24 @@ import * as KV from '../../kodeverk';
- * når det asynkrone kallet, feks fra API'et er ferdigkjørt.
- *
- */
+const lagState = data => ({ behandlingsgrunnlag: { data } });
```
~~Kan vel bruke test-utils sin lagState i stedet for denne? Dog blir det litt rart å hente fra en modul som heter "test.." her...~~ Du kan sikkert beholde det slik det er

JohanSakshaug commented on src/felleskomponenter/paneler/andreArbeidsforholdNorge/andreArbeidsforholdNorge.js:
```diff
@@ -17,71 +16,63 @@ import ArbeidsforholdNorgeListe from './arbeidsforholdNorgeListe';
+  const finnOrganisasjon = orgnr => {
+    if (organisasjoner.map(o => o.orgnr).includes(orgnr)) {
+      return organisasjoner.find(o => o.orgnr === orgnr);
+    }
```
Denne kan kanskje forenkles litt:
`const org = organisasjoner.find(o => o.orgnr === orgnr);`
`return org || { orgnr };`

JohanSakshaug commented on src/ducks/behandlingsgrunnlag/operations.js:
```diff
@@ -13,22 +13,24 @@ import { formSelectors } from '../form';
-    }
-  );
+  return (dispatch, getState) => {
+    const f = doThenDispatch(
```
Siden `doThenDispatch` returnerer en thunk:
`const f` -> `const thunk`?

sigbjorn-myhre committed:
QA MELOSYS-4001

sigbjorn-myhre committed:
Merge branch 'develop' into MELOSYS-4001_vis_norske_arbeidsgivere_fra_sed
